### PR TITLE
Support London hardfork

### PR DESCRIPTION
**Pull request by vgeddes**
Specifically these EIPs which are already active on Ropsten, and will be rolled out on Mainnet on the 4th August:
- [EIP-3554](https://eips.ethereum.org/EIPS/eip-3554): Difficulty Bomb Delay to December 2021
- [EIP-1554](https://eips.ethereum.org/EIPS/eip-1559): Fee market changes

Changes:
- [x] EIP-3554: Adjust the difficulty verifier in the light client.
- [x] EIP-1554: Added an optional `base_fee_for_gas` field to the header and updating the hashing method accordingly.
- [x] Various tweaks and improvements to E2E tests
- [x] Update incorrect block weight parameters in parachain runtimes 
- [x] Update ethashproof header seal hash: https://github.com/Snowfork/ethashproof/pull/4
- [x] Update unit tests

The changes are backwards compatible in order to work with Ganache which hasn't been updated yet.

Testing:
* With Ropsten, ethereum light client successfully verifies and imports headers
* With local Ganache (pre London hardfork), E2E tests pass

### CODE REVIEW

File: ethereum/deploy/01-beefy.ts [hunk 1]
```diff
@@ -21,11 +21,13 @@ module.exports = async ({
         MerkleProof: merkleProofLibrary.address
     },
     log: true,
+    autoMine: true,
```
Comment (musnit) [inline ethereum/deploy/01-beefy.ts:24]: does this field just get ignored for ropsten deployments? if so, then great

File: parachain/runtime/local/src/lib.rs [hunk 2]
```diff
@@ -169,11 +169,11 @@ const MAXIMUM_BLOCK_WEIGHT: Weight = WEIGHT_PER_SECOND / 2;
 parameter_types! {
 	pub const Version: RuntimeVersion = VERSION;
 	pub const BlockHashCount: BlockNumber = 2400;
-	/// We allow for 2 seconds of compute with a 6 second average block time.
+	/// We allow for 0.5 seconds of compute with a 12 second average block time.
```
Comment (musnit) [inline parachain/runtime/local/src/lib.rs:172]: hm mind explaining what this is for or linking to a relevant substrate doc where i can read about it?

File: ethereum/deploy/01-beefy.ts (hunk 1 above)
Comment (vgeddes) [reply ethereum/deploy/01-beefy.ts:24]: yeah :+1:

File: test/package.json [hunk 3]
```diff
@@ -4,15 +4,13 @@
   "main": "index.js",
   "license": "MIT",
   "scripts": {
-    "test": "mocha --timeout 210000 --exit"
+    "test": "mocha --timeout 300000 --exit"
```
Comment (musnit) [inline test/package.json:7]: 😭 keeps getting worse lol

File: test/scripts/start-services.sh [hunk 4]
```diff
@@ -15,27 +15,7 @@ start_ganache()
 
     npx ganache-cli \
         --port=8545 \
-        --networkId=344 \
-        --chainId=344 \
-        --deterministic \
-        --db $configdir/ganache.db \
-        --mnemonic="stone speak what ritual switch pigeon weird dutch burst shaft nature shove" \
-        --gasLimit=8000000 \
```
Comment (musnit) [inline test/scripts/start-services.sh:23]: nice change with the automine, thx!

File: parachain/runtime/local/src/lib.rs (hunk 2 above)
Comment (vgeddes) [reply parachain/runtime/local/src/lib.rs:172]: Yeah so while going some rabbit-holes while testing header verification (that's a story for another time), I discovered that the previous settings were incorrect and could lead to relay chain validators failing to validate our blocks. Basically the maximum block weight of a parachain block is bounded by the maximum weight of relay chain block (0.5s of weight).

I copied this over from statemint/rococo parachains, as its a more realistic figure: https://github.com/paritytech/cumulus/blob/master/polkadot-parachains/statemint-common/src/lib.rs#L86

Strictly speaking this change isn't relevant to Ropsten or the London hardfork. Only made it while troubleshooting an issue.

File: test/package.json (hunk 3 above)
Comment (vgeddes) [reply test/package.json:7]: I may be able to revert this. Was tweaking a bunch of settings while troubleshooting.

**Comment by vgeddes** [comment]
> looks good. if e2e tests pass then i think can merge

yeah e2e and unit tests all pass now. Merging.

**Review by philipstanislaus — COMMENTED** [review]
LGTM :pray

**Pull request merged**